src/test/java/nc/students/ayaz/repositories/UserRepositoryTest.java

```diff
@@ -0,0 +1,34 @@
+package nc.students.ayaz.repositories;
+
+import nc.students.ayaz.model.User;
+import nc.students.ayaz.model.exceptions.NoSuchUserException;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.junit.Assert.assertEquals;
+
+public class UserRepositoryTest {
+
+    @Rule
+    public ExpectedException exception = ExpectedException.none();
+
+    @Test
+    public void shouldGetUserByNicknameWhenAddedEarlier() throws NoSuchUserException {
+        UserRepository repository = new UserRepository();
+        User user = new User("Ayaz");
+        repository.addUser(user);
+
+        User fetchedUser = repository.getUserByNickname("Ayaz");
+
+        assertEquals(user, fetchedUser);
+    }
+
+    @Test
+    public void shouldThrowNoSuchUserExceptionWhenRepositoryDoesntHaveUser() throws NoSuchUserException {
+        UserRepository repository = new UserRepository();
+        exception.expect(NoSuchUserException.class);
```

Comment:
&#x1F34E; **Issue**
Its clearer to use @Test(expect = NoSuchUserException.class) instead of ExpectedException.
As I know you should use ExpectedException only when you want to make very specific assertions on Exception that got thrown.
If you just want to make sure that exception got thrown, you will get more readable code without ExpectedException.